### DISCUSSION TITLE
Followed Getting Started But Keep Getting "repository-not-found"

### DISCUSSION THREAD

**Discussion by mikecann** [avitorio/outstatic#159] [Q&A] [answered]
I followed the [getting started guide](https://outstatic.com/docs/getting-started) to the T.

I have an OAuth app correctly configured:

![image](https://github.com/avitorio/outstatic/assets/215033/44b68ffd-1529-46d2-b00a-6d045edda282)

![image](https://github.com/avitorio/outstatic/assets/215033/756cd07e-bbe7-4a55-81ab-3275cb075b76)

I have the site setup on vercel perfectly:

![image](https://github.com/avitorio/outstatic/assets/215033/83120bc7-0b4c-42e1-84e4-c41c728bdab7)

https://pgloa-website.vercel.app/

But when I attempt to signin with github at this URL: https://pgloa-website.vercel.app/outstatic I get the following error:

![image](https://github.com/avitorio/outstatic/assets/215033/187f11c5-c8b4-46d8-b508-f924fb7515ea)

Url of that page is: https://pgloa-website.vercel.app/outstatic?error=repository-not-found

I followed the troubleshooting guide and am doing everything they say: https://outstatic.com/docs/faqs#troubleshooting-login-and-repository-access-issues

I even set the repo public (which im sure I shouldnt need to do): https://github.com/mikecann/pgloa-website

Any ideas?

**Comment by avitorio** [comment]
Hey @mikecann,

Thanks for trying Outstatic! I'm sorry to hear you are having trouble with your install.
I want to make sure your env vars are correct.

For your project it should be:

```
OST_REPO_SLUG=pgloa-website
OST_REPO_OWNER=mikecann
```

Is that what you have? Let me know.

**Reply by avitorio** [reply]
I'm asking because that error specify that the repository was not found. Are you trying to log in with a second github account?
If that's the case you'll need to specify the OST_REPO_OWNER.

If not, can you add me to the repo? I'll take a look at it.

**Reply by mikecann** [reply, marked as answer]
I added the `OST_REPO_OWNER` and no change.

Then I had a thought overnight, maybe I need to re-deploy on vercel.

I did so and now it works!??

Haha maybe I should have known that, maybe it should be in the docs *shrug* anyways, at least it works now :) thanks!

**Reply by avitorio** [reply]
Glad to hear you figured it out!
I'll make sure to add a reminder to restart your server or redeploy when updating environment variables to the docs. Cheers!

**Reply by nmcpmadmin** [reply]
I had this exact same issue. The deployment doesn't pick up the changes until it is redeployed. 
I've done this and it works for me too.

**Reply by avitorio** [reply]
Hey @nmcpmadmin and @mikecann ,
Thanks for this feedback.

Where do you think it makes sense for us to add this notice in the docs?

**Comment by nmcpmadmin** [comment]
Hi Andre,

Good question; I think it needs a single line of text on the FAQs, right
after the part about variables for slug - ‘With all the settings given,
reploy your site to update the changes’ (or, something like that).

Matt

**Reply by avitorio** [reply]
[Done](https://outstatic.com/docs/faqs#troubleshooting-login-and-repository-access-issues)! Thank you for the suggestion.

**Comment by nmcpmadmin** [comment]
Excellent, Andre - that should help anyone else who encounters that issue.
Have a good week.

Matt


.

**Comment by tailorvj** [comment]
@avitorio I've had a similar issue in my development environment on Gitpod.io. I commented out all of the settings in .env.local, except for the first 4 settings, restarted the development server and managed to login. The problem is that the redirect sent me back to localhost:3000 instead of to the app URL defined in Github OAuth apps. I guess it must be hardcoded somewhere. The login was successful though, so when I opened the actual <gitpodurl>/outstatic URL, I was logged in.

I have 2 suggestions:

1. The .env.local.example should have everything commented out by default, except for the first 4 settings
2. The redirect URL for local development shouldn't be hardcoded, but instead use the URL setting from the Github OAuth App settings during the login process

You can try this yourself by adding the following .gitpod.yml file to the root of the blog project and opening the Github repo on Gitpod.io:

```yml
# List the start up tasks. Learn more https://www.gitpod.io/docs/config-start-tasks/
tasks:
  - name: Start development environment
    init: npm install
    command: npm run dev

# List the ports to expose. Learn more https://www.gitpod.io/docs/config-ports/
ports:
  - port: 3000
    onOpen: open-preview
    description: Website preview
```

Kind regards,
Tailor

**Reply by avitorio** [reply]
Hi @tailorvj - thanks for the suggestions.

I'll look into this asap.

It's weird that it is redirecting to localhost... I don't believe that is hardcoded anywhere on the package. It should redirect to whatever the callback url is on GitHub. I'll double check.
